content/shmem_request_wait.tex

```diff
@@ -0,0 +1,34 @@
+\apisummary{
+    The request wait routine provides a method for completing
+    outstanding operations on a request.
+}
+
+\begin{apidefinition}
+
+\begin{Csynopsis}
+void shmem_request_wait(shmem_request_t *req);
+\end{Csynopsis}
+
+\begin{apiarguments}
+    \apiargument{IN}{req}{Handle to the request object.}
+\end{apiarguments}
+
+\apidescription{
+    The \FUNC{shmem\_request\_wait} routine blocks until all outstanding operations on the request
+    are finished. Outstanding operations are operations posted before
+    wait was called.
+    There are no guarantees on the ordering of the outstanding
+    operations.
+}
+
+\apireturnvalues{
+    None.
+}
+
+\apinotes{ In a threaded environment \FUNC{shmem\_request\_wait} from
+a thread other than the thread, posting the operation is permitted.
+However, only one thread can call \FUNC{shmem\_request\_wait} for a
+particular request.
+}
```

Comment:
Notes from WG discussion: The paragraph should be reworked to reflect the ordering requirements between the operations and wait.